### DISCUSSION TITLE
Future Plans?

### DISCUSSION THREAD

**Discussion by sfranzyshen** [cesanta/elk#37] [Q&A] [answered]
I am curious if there are any plans to expand the functionality (add more ES6 support) to Elk? Or, will the expansion be frozen to preserve a smaller size? 

For example, "[ternary operator](https://github.com/cesanta/elk/commit/bcaeba7ff20f4bad2c73f8913327371af1cdd109)" was briefly added (preliminary) and then removed in a [later pull](https://github.com/cesanta/elk/commit/37eefc3457084a7291a0caa9d31afbccc5302ecf) with a note "optimizations to save flash space" 

Will "Smaller Size" take precedent over "Functionality" ?

**Comment by cpq** [comment, marked as answer]
Yeah, small size and simplicity takes the highest priority.
Elk implements a very minimum required to have an ability to script.
Adding more features is a never ending process and arguments about what is useful vs what is not.
Elk is going to be expanded only if there is a very obvious use case, on a real-life embedded example, where a feature is missing, and it is impossible or very hard to implement a functionality using an existing set of features.
